endpoints/openrtb2/auction_test.go

```diff
@@ -1145,6 +1146,75 @@ func TestStoredRequests(t *testing.T) {
 	}
 }
 
+func TestValidateOrFillChannel(t *testing.T) {
+	testCases := []struct {
+		description           string
+		givenIsAmp            bool
+		givenRequestWrapper   *openrtb_ext.RequestWrapper
+		expectedError         error
+		expectedChannelObject *openrtb_ext.ExtRequestPrebidChannel
+	}{
+		{
+			description: "No request.ext info in app request, so we expect channel name to be set to app",
+			givenRequestWrapper: &openrtb_ext.RequestWrapper{
+				BidRequest: &openrtb2.BidRequest{App: &openrtb2.App{}},
+			},
```

Comment:
Please also add `app` and `amp` test cases where `request.ext` exists but `request.ext.prebid` doesn't exist and where `request.ext.prebid` exists but the channel object doesn't exist